src/context/WalletProvider/KeepKey/hooks/useKeepKeyEventHandler.ts

```diff
@@ -95,23 +105,39 @@ export const useKeepKeyEventHandler = (state: KeyringState, dispatch: Dispatch<A
       try {
         const id = keyring.getAlias(deviceId)
         if (id === state.walletInfo?.deviceId) {
-          dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: false })
+          dispatchWallet({ type: WalletActions.SET_IS_CONNECTED, payload: false })
         }
       } catch (e) {
         console.error('Device Disconnect Error:', e)
       }
     }
 
+    const setAwaitingButtonPress = (activeRequest: boolean) => {
+      dispatchKeepKey({
+        type: KeepKeyActions.SET_AWAITING_BUTTON_PRESS,
+        payload: activeRequest
+      })
+    }
+
+    const updateStatus = (status: UpdateStatus) => {
+      dispatchKeepKey({
+        type: KeepKeyActions.SET_UPDATE_STATUS,
+        payload: status
+      })
+    }
+
```

Comment:
I'd move these two action creators in `<KeepKeyProvider />`, next to the `reset` action creator:
https://github.com/shapeshift/web/pull/1341/commits/3a6c512aab2eaf53db09f2eb8deab9da5bfb8ce2#diff-81fd9ebe2a1638c6ea22ea8a13af9bbc5476682929f3cfb42a4f8e1c1aa11391R50

Reply:
Fantastic suggestion, thank you ser: https://github.com/shapeshift/web/pull/1341/commits/244c943fc559b7d1cbe97b2b7f1fd64b77275e3f